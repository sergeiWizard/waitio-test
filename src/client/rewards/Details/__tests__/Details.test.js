import React from 'react';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';
import { mountWithIntl } from 'enzyme-react-intl';
import { MemoryRouter as Router } from 'react-router-dom';
import Details from '../Details';
import {
  initialState,
  stateForActiveReserve,
  stateForFrequency,
  stateForIsEligible,
  stateForPostRequirements,
  stateForReservePastPress,
} from '../__mock__/mockData';

jest.mock('../../CampaignCardHeader/CampaignCardHeader', () => () => (
  <div className="CampaignCardHeader" />
));

const toggleModal = jest.fn();
const reserveOnClickHandler = jest.fn();

describe('Details', () => {
  let wrapper;

  beforeEach(() => {
    const props = initialState;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
  });

  afterEach(() => jest.clearAllMocks());

  it('renders and matches snapshot', () => {
    const props = initialState;
    wrapper = shallow(
      <IntlProvider>
        <Details {...props} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should find class Details__text-wrap', () => {
    const container = wrapper.find('.Details__text-wrap');
    expect(container).toHaveLength(1);
  });

  it('should find class Details__text', () => {
    const container = wrapper.find('.Details__text');
    expect(container).toHaveLength(7);
  });

  it('should find class Details__criteria-wrap', () => {
    const container = wrapper.find('.Details__criteria-wrap');
    expect(container).toHaveLength(2);
  });

  it('should find class Details__criteria-row', () => {
    const container = wrapper.find('.Details__criteria-row');
    expect(container).toHaveLength(8);
  });

  it('should find class nowrap', () => {
    const container = wrapper.find('.nowrap');
    expect(container).toHaveLength(2);
  });

  it('should find class no-visible', () => {
    const container = wrapper.find('.no-visible');
    expect(container).toHaveLength(2);
  });

  it('should find class Details__footer', () => {
    const container = wrapper.find('.Details__footer');
    expect(container).toHaveLength(1);
  });

  it('should find class Details__footer-reserve-btn', () => {
    const container = wrapper.find('.Details__footer-reserve-btn');
    expect(container).toHaveLength(1);
  });

  it('should find class ant-btn', () => {
    const container = wrapper.find('.ant-btn');
    expect(container).toHaveLength(2);
  });

  it('should find in button attribute "disabled"', () => {
    expect(wrapper.find('.ant-btn-primary').props().disabled).toBe(true);
  });

  it('should be "active" button "Reserved"', () => {
    const props = stateForActiveReserve;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    wrapper.find('.ant-btn-primary').simulate('click');
    expect(reserveOnClickHandler).toHaveBeenCalled();
  });

  it('should get link www.waivio.com/object/testnet', () => {
    const textPayables = wrapper.find('.Details__criteria-link').first();
    expect(textPayables.text()).toEqual('www.waivio.com/object/testnet');
  });

  it('should get link www.waivio.com/object/testing123', () => {
    const textPayables = wrapper.find('.Details__criteria-link').last();
    expect(textPayables.text()).toEqual('www.waivio.com/object/testing123');
  });

  it('should be "active" button "Write review"', () => {
    const props = stateForReservePastPress;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const btn = wrapper.find('.ant-btn-primary');
    expect(btn.text()).toEqual('Write review');
  });

  it('should be add new checkbox"', () => {
    const props = stateForFrequency;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const container = wrapper.find('.Details__criteria-row').at(3);
    expect(container.text()).toEqual(
      'Have not received a reward from @alexeygrigurko for reviewing Testing123 in the last 1 days and does not have an active reservation for such a reward at the moment.',
    );
  });

  it('should get text Minimum Waivio expertise: 1', () => {
    const props = initialState;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const textPayables = wrapper.find('.Details__criteria-row').at(0);
    expect(textPayables.text()).toEqual('Minimum Waivio expertise: 1');
  });

  it('should get text Minimum number of followers: 2', () => {
    const props = initialState;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const textPayables = wrapper.find('.Details__criteria-row').at(1);
    expect(textPayables.text()).toEqual('Minimum number of followers: 2');
  });

  it('should get text Minimum number of posts: 3', () => {
    const props = initialState;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const textPayables = wrapper.find('.Details__criteria-row').at(2);
    expect(textPayables.text()).toEqual('Minimum number of posts: 3');
  });

  it('should simulate button "Cancel', () => {
    const props = initialState;

    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    wrapper
      .find('.ant-btn')
      .first()
      .simulate('click');
    expect(toggleModal).toHaveBeenCalledTimes(1);
  });

  it('should find in isEligible attribute "disabled"', () => {
    const props = stateForIsEligible;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    expect(wrapper.find('.ant-btn-primary').props().disabled).toBe(true);
  });

  it('should found in Post requirements new item', () => {
    const props = stateForPostRequirements;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const container = wrapper.find('.Details__criteria-row').at(6);
    expect(container.text()).toEqual('2. Photo of the receipt (without personal details);');
  });

  it('should found in Reward bot "aggroed"', () => {
    const props = stateForPostRequirements;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const container = wrapper.find('a').at(9);
    expect(container.props().href).toBe('/object/aggroed');
  });

  it('should found in Reward bot "fenrir78"', () => {
    const props = stateForPostRequirements;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const container = wrapper.find('a').at(10);
    expect(container.props().href).toBe('/object/fenrir78');
  });

  it('should found in Reward bot "ace108"', () => {
    const props = stateForPostRequirements;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const container = wrapper.find('a').at(11);
    expect(container.props().href).toBe('/object/ace108');
  });

  it('should return link Terms', () => {
    const props = stateForPostRequirements;
    wrapper = mountWithIntl(
      <Router>
        <Details
          toggleModal={toggleModal}
          reserveOnClickHandler={reserveOnClickHandler}
          {...props}
        />
      </Router>,
    );
    const container = wrapper.find('a').at(12);
    expect(container.props().href).toBe('/object/xrj-terms-and-conditions/page');
  });
});
