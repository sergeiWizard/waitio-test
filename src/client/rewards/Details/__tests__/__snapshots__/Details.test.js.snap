// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <InjectIntl(Details)
      assigned={null}
      isModalDetailsOpen={true}
      isReserved={false}
      isReviewDetails={false}
      loading={false}
      objectDetails={
        Object {
          "__v": 0,
          "_id": "5e5e6a33fb97af17249e3f7f",
          "activation_permlink": "activate-monterey-7y5q4n886t5",
          "agreementObjects": Array [],
          "app": "waiviodev",
          "blacklist_users": Array [],
          "budget": 1,
          "commissionAgreement": 0.05,
          "compensationAccount": "",
          "count_objects": 1,
          "count_reservation_days": 7,
          "count_users": 0,
          "createdAt": "2020-03-03T14:31:15.970Z",
          "existAssigns": Array [],
          "expired_at": "2020-03-07T14:31:11.877Z",
          "frequency_assign": 0,
          "frequency_exceed": false,
          "guide": Object {
            "alias": "Alex Grigurko",
            "name": "alexeygrigurko",
            "totalPayed": 1.819,
            "wobjects_weight": 1496742.3025625013,
          },
          "is_sponsor_blacklist": false,
          "match_bots": Array [],
          "max_assign_count": 1,
          "name": "Testing123",
          "objects": Array [
            Object {
              "assigned": null,
              "count_users": 0,
              "object": Object {},
              "permlink": null,
            },
          ],
          "payments": Array [],
          "requiredObject": "testing123",
          "required_object": Object {},
          "requirement_filters": Object {
            "can_assign_by_budget": true,
            "can_assign_by_current_day": true,
            "expertise": false,
            "followers": true,
            "freeReservation": true,
            "frequency": true,
            "not_blacklisted": true,
            "posts": true,
          },
          "requirements": Object {
            "minPhotos": 0,
            "receiptPhoto": false,
          },
          "reservation_timetable": Object {},
          "reserved_users": Array [],
          "reward": 0.001,
          "sponsors_blacklist": Array [],
          "status": "active",
          "type": "reviews",
          "updatedAt": "2020-03-03T14:31:25.908Z",
          "userRequirements": Object {
            "minExpertise": 1,
            "minFollowers": 2,
            "minPosts": 3,
          },
          "usersLegalNotice": "",
          "whitelist_users": Array [],
        }
      }
      proposedWobj={
        Object {
          "__v": 0,
          "_id": "5cf76fdb9619f621832a05ca",
          "app": "waiviodev/1.0.0",
          "author": "x6oc5",
          "author_permlink": "testnet",
        }
      }
      requiredObjectName="testing123"
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "assigned": null,
      "isModalDetailsOpen": true,
      "isReserved": false,
      "isReviewDetails": false,
      "loading": false,
      "objectDetails": Object {
        "__v": 0,
        "_id": "5e5e6a33fb97af17249e3f7f",
        "activation_permlink": "activate-monterey-7y5q4n886t5",
        "agreementObjects": Array [],
        "app": "waiviodev",
        "blacklist_users": Array [],
        "budget": 1,
        "commissionAgreement": 0.05,
        "compensationAccount": "",
        "count_objects": 1,
        "count_reservation_days": 7,
        "count_users": 0,
        "createdAt": "2020-03-03T14:31:15.970Z",
        "existAssigns": Array [],
        "expired_at": "2020-03-07T14:31:11.877Z",
        "frequency_assign": 0,
        "frequency_exceed": false,
        "guide": Object {
          "alias": "Alex Grigurko",
          "name": "alexeygrigurko",
          "totalPayed": 1.819,
          "wobjects_weight": 1496742.3025625013,
        },
        "is_sponsor_blacklist": false,
        "match_bots": Array [],
        "max_assign_count": 1,
        "name": "Testing123",
        "objects": Array [
          Object {
            "assigned": null,
            "count_users": 0,
            "object": Object {},
            "permlink": null,
          },
        ],
        "payments": Array [],
        "requiredObject": "testing123",
        "required_object": Object {},
        "requirement_filters": Object {
          "can_assign_by_budget": true,
          "can_assign_by_current_day": true,
          "expertise": false,
          "followers": true,
          "freeReservation": true,
          "frequency": true,
          "not_blacklisted": true,
          "posts": true,
        },
        "requirements": Object {
          "minPhotos": 0,
          "receiptPhoto": false,
        },
        "reservation_timetable": Object {},
        "reserved_users": Array [],
        "reward": 0.001,
        "sponsors_blacklist": Array [],
        "status": "active",
        "type": "reviews",
        "updatedAt": "2020-03-03T14:31:25.908Z",
        "userRequirements": Object {
          "minExpertise": 1,
          "minFollowers": 2,
          "minPosts": 3,
        },
        "usersLegalNotice": "",
        "whitelist_users": Array [],
      },
      "proposedWobj": Object {
        "__v": 0,
        "_id": "5cf76fdb9619f621832a05ca",
        "app": "waiviodev/1.0.0",
        "author": "x6oc5",
        "author_permlink": "testnet",
      },
      "requiredObjectName": "testing123",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "assigned": null,
        "isModalDetailsOpen": true,
        "isReserved": false,
        "isReviewDetails": false,
        "loading": false,
        "objectDetails": Object {
          "__v": 0,
          "_id": "5e5e6a33fb97af17249e3f7f",
          "activation_permlink": "activate-monterey-7y5q4n886t5",
          "agreementObjects": Array [],
          "app": "waiviodev",
          "blacklist_users": Array [],
          "budget": 1,
          "commissionAgreement": 0.05,
          "compensationAccount": "",
          "count_objects": 1,
          "count_reservation_days": 7,
          "count_users": 0,
          "createdAt": "2020-03-03T14:31:15.970Z",
          "existAssigns": Array [],
          "expired_at": "2020-03-07T14:31:11.877Z",
          "frequency_assign": 0,
          "frequency_exceed": false,
          "guide": Object {
            "alias": "Alex Grigurko",
            "name": "alexeygrigurko",
            "totalPayed": 1.819,
            "wobjects_weight": 1496742.3025625013,
          },
          "is_sponsor_blacklist": false,
          "match_bots": Array [],
          "max_assign_count": 1,
          "name": "Testing123",
          "objects": Array [
            Object {
              "assigned": null,
              "count_users": 0,
              "object": Object {},
              "permlink": null,
            },
          ],
          "payments": Array [],
          "requiredObject": "testing123",
          "required_object": Object {},
          "requirement_filters": Object {
            "can_assign_by_budget": true,
            "can_assign_by_current_day": true,
            "expertise": false,
            "followers": true,
            "freeReservation": true,
            "frequency": true,
            "not_blacklisted": true,
            "posts": true,
          },
          "requirements": Object {
            "minPhotos": 0,
            "receiptPhoto": false,
          },
          "reservation_timetable": Object {},
          "reserved_users": Array [],
          "reward": 0.001,
          "sponsors_blacklist": Array [],
          "status": "active",
          "type": "reviews",
          "updatedAt": "2020-03-03T14:31:25.908Z",
          "userRequirements": Object {
            "minExpertise": 1,
            "minFollowers": 2,
            "minPosts": 3,
          },
          "usersLegalNotice": "",
          "whitelist_users": Array [],
        },
        "proposedWobj": Object {
          "__v": 0,
          "_id": "5cf76fdb9619f621832a05ca",
          "app": "waiviodev/1.0.0",
          "author": "x6oc5",
          "author_permlink": "testnet",
        },
        "requiredObjectName": "testing123",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;
